src/umpire/strategy/MixedPool.inl

```diff
@@ -0,0 +1,157 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/util/Macros.hpp"
+#include "umpire/strategy/MixedPool.hpp"
+
+namespace umpire {
+namespace strategy {
+
+namespace {
+
+template<int FirstFixed, int Current, int LastFixed, int Increment>
+struct make_fixed_pool_array {
+  template<typename Array>
+  static void eval(Array& fixed_pool, Allocator& allocator) {
+    const int index = Current - FirstFixed;
+    const std::size_t size = 1 << Current;
+    std::stringstream ss{"internal_fixed_"};
+    ss << size;
+    fixed_pool[index] = std::make_shared<FixedPool<unsigned char[size]> >(ss.str(), -1, allocator);
+    make_fixed_pool_array<FirstFixed, Current+Increment, LastFixed, Increment>::eval(fixed_pool, allocator);
+  }
+};
+
+template<int FirstFixed, int LastFixed, int Increment>
+struct make_fixed_pool_array<FirstFixed,LastFixed,LastFixed,Increment> {
+  template<typename Array>
+  static void eval(Array& fixed_pool, Allocator& allocator) {
+    const int index = LastFixed - FirstFixed;
+    const std::size_t size = 1 << LastFixed;
+    std::stringstream ss{"internal_fixed_"};
+    ss << size;
+    fixed_pool[index] = std::make_shared<FixedPool<unsigned char[size]> >(ss.str(), -1, allocator);
+  }
+};
+
+} // anonymous namespace
+
+inline static size_t nextPower2(unsigned int n) {
```

Comment:
I suggest that either a comment is added or the name is updated to reflect that this will only work for a 32 bit number

Reply:
Good idea!